pkg/authorization/registry/selfsubjectrulesreview/storage.go

```diff
@@ -22,12 +28,48 @@ func (r *REST) New() runtime.Object {
 
 // Create registers a given new ResourceAccessReview instance to r.registry.
 func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
-	// the input object has no valuable input, so don't bother checking it.false
+	rulesReview, ok := obj.(*authorizationapi.SelfSubjectRulesReview)
+	if !ok {
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("not a SelfSubjectRulesReview: %#v", obj))
+	}
+	namespace := kapi.NamespaceValue(ctx)
+	if len(namespace) == 0 {
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("namespace is required on this type: %v", namespace))
+	}
+	user, exists := kapi.UserFrom(ctx)
+	if !exists {
+		return nil, fmt.Errorf("user missing from context")
+	}
+
 	policyRules, err := r.ruleResolver.GetEffectivePolicyRules(ctx)
+	rules := []authorizationapi.PolicyRule{}
+	for _, rule := range policyRules {
+		rules = append(rules, rulevalidation.BreakdownRule(rule)...)
```

Comment:
> actually, doesn't the Covers check already break down the rules? just wondering how expensive this API call is going to be

It breaks down the needle, but if the needle is already broken down, that's a reasonably fast no-op.  No net allocations compared to the alternative, maybe one extra pass through a loop?